website/payments/services.py

```diff
@@ -189,3 +189,16 @@ def send_tpay_batch_processing_emails(batch):
             },
         )
     return len(member_payments)
+
+
+def execute_data_minimization(dry_run=False):
+    """Anonymizes payments older than 7 years."""
+    # Sometimes years are 366 days of course, but better delete 1 or 2 days early than late
+    deletion_period = timezone.now().date() - timezone.timedelta(days=(365 * 7))
+
+    queryset = Payment.objects.filter(created_at__lte=deletion_period).exclude(
+        paid_by__isnull=True
+    )
+    if not dry_run:
+        queryset.update(paid_by=None)
+    return queryset.all()
```

Comment:
```suggestion
    return queryset
```

don't think you need that